tests/unit/project_loader/test_parts.py

```diff
@@ -138,3 +138,71 @@ def test_load_with_dummy_repo(self, os_release_mock):
         # Ensure the dummy repo returns an empty set for required
         # build tools.
         project_loader.load_config()
+
+
+class PartOrderTestCase(unit.TestCase):
+
+    scenarios = [
+        ('part1 then part2', {
+            'contents': dedent("""\
+                name: test
+                version: "1"
+                summary: test
+                description: test
+                confinement: strict
+
+                parts:
+                  part1:
+                    plugin: nil
+                  part2:
+                    plugin: nil
+                """),
+            'expected_order': ['part1', 'part2'],
+        }),
+        ('part2 then part1', {
+            'contents': dedent("""\
+                name: test
+                version: "1"
+                summary: test
+                description: test
+                confinement: strict
+
+                parts:
+                  part2:
+                    plugin: nil
+                  part1:
+                    plugin: nil
+                """),
+            'expected_order': ['part1', 'part2'],
+        }),
+        ('after', {
+            'contents': dedent("""\
+                name: test
+                version: "1"
+                summary: test
+                description: test
+                confinement: strict
+
+                parts:
+                  part2:
+                    plugin: nil
+                    after: [part3]
+                  part1:
+                    plugin: nil
+                    after: [part3]
```

Comment:
can I see a test with only `part2` having `after: [part3]`?

Reply:
Of course, done.